libpromises/locks.c

```diff
@@ -554,6 +554,33 @@ static void PromiseTypeString(char *dst, size_t dst_size, const Promise *pp)
     }
 }
 
+// returns true if pid points to a cfengine process
+static bool IsCfengineProcess(const int pid)
+{
+    // There is no /proc on windows
+#ifndef __linux__
+    char procfile[PATH_MAX];
+    char cmd[PATH_MAX]; // we don't need the full ARG_MAX
+    snprintf(procfile, PATH_MAX, "/proc/%d/cmdline", pid);
+    FILE *f = fopen(procfile, "r");
+    if (f != NULL){
+        size_t size;
+        size = fread(cmd, sizeof(char), PATH_MAX, f);
+        if (size > 0){
+            if ('\n' == cmd[size-1]) {
+                cmd[size-1] = '\0';
+            }
+        }
+        fclose(f);
+        // cmd contains the process command path
+        char *name = basename(cmd);
+        return (strncmp("cf-", name, 3) == 0);
+    }
+#endif
+    // assume true if we don't know
+    return true;
```

Comment:
Wouldn't it be safer to assume the opposite?

Reply:
we don't know means use the old behaviour on other platforms